tests/test_regular_coloring.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2024 D-Wave Systems Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import unittest
+
+import dwave_networkx as dnx
+import numpy as np
+
+class TestRegularColoring(unittest.TestCase):
+
+    def test_valid(self):
+        test_cases = {'chimera': [(3,3), (4,4)],
+                      'pegasus': [(2,), (4,)],
+                      'zephyr-0': [(1,2), (3,4)],
+                      'zephyr-1': [(1,3), (3,2)]}
+
+        for topology_type, topology_shapes in test_cases.items():
```

Comment:
Good candidate for `parameterized`, if you feel like simplifying. For an example, you can check [here](https://github.com/dwavesystems/dwave-cloud-client/blob/b2432fe441d304162ee2f3bac6249241366ece91/tests/auth/test_server.py#L32-L42).

Reply:
Next pull request